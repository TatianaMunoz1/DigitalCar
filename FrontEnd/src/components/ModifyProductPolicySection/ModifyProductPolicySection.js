import React, { useState, useEffect } from "react";
import { useHistory } from "react-router";
import Button from "../Button/Button";
import styles from "./ModifyProductPolicySection.module.css";
import Swal from "sweetalert2";
import axios from "axios";
import Url from "../../Url"

function ModifyProductPolicySection(props) {
  const history = useHistory();
  const [submit, setSubmit] = useState(null);
  const [productPolicy, setProductPolicy] = useState({
    deliveryStandards: "",
    healthAndSecurity: "",
    cancellationPolicy: "",
  });

  useEffect(() => {
    axios
      .get(Url+"/politics")
      .then((response) => {
        if (response.data !== "null")
          setProductPolicy({
            deliveryStandards: response.data.delivery,
            healthAndSecurity: response.data.healthAndSecurity,
            cancellationPolicy: response.data.cancellation,
          });
      })
      .catch((error) => {});
  }, []);

  const handleInputChange = (e) => {
    const target = e.target;
    const value = target.value;
    const name = target.name;

    setProductPolicy({
      ...productPolicy,
      [name]: value,
    });
  };

  const onSubmit = (e) => {
    e.preventDefault();
    setSubmit(true);

    const data = {
      delivery: `${productPolicy.deliveryStandards}`,
      healthAndSecurity: `${productPolicy.healthAndSecurity}`,
      cancellation: `${productPolicy.cancellationPolicy}`,
    };

    if (
      productPolicy.deliveryStandards !== "" &&
      productPolicy.healthAndSecurity !== "" &&
      productPolicy.cancellationPolicy !== ""
    ) {
      const config = {
        headers: { Authorization: props.user.token },
      };

      axios
        .put("http://localhost:8080/politics", data, config)
        .then(function (response) {
          if (response.status === 200) {
            Swal.fire({
              icon: "success",
              title: "Se actualizaron las políticas del producto.",
              showDenyButton: true,
              confirmButtonText: "Ir al home",
            }).then((result) => {
              if (result.isConfirmed) {
                history.push("/");
              }
            });
          } else {
            const Toast = Swal.mixin({
              toast: true,
              position: "top-end",
              showConfirmButton: false,
              background: "#ffc2c2",
              timer: 5000,
              timerProgressBar: true,
              didOpen: (toast) => {
                toast.addEventListener("mouseenter", Swal.stopTimer);
                toast.addEventListener("mouseleave", Swal.resumeTimer);
              },
            });
            Toast.fire({
              icon: "error",
              title:
                "<h4 style='color:#B00020'>Lamentablemente el producto no ha podido crearse. </br>Por favor intente más tarde.</h4>",
            });
          }
        })
        .catch(function (error) {
          const Toast = Swal.mixin({
            toast: true,
            position: "top-end",
            showConfirmButton: false,
            background: "#ffc2c2",
            timer: 5000,
            timerProgressBar: true,
            didOpen: (toast) => {
              toast.addEventListener("mouseenter", Swal.stopTimer);
              toast.addEventListener("mouseleave", Swal.resumeTimer);
            },
          });
          Toast.fire({
            icon: "error",
            title:
              "<h4 style='color:#B00020'>Lamentablemente el producto no ha podido crearse. </br>Por favor intente más tarde.</h4>",
          });
        });
    }
  };

  return (
    <div>
      <form className={styles.container}>
        {(productPolicy.deliveryStandards === "" ||
          productPolicy.healthAndSecurity === "" ||
          productPolicy.cancellationPolicy === "") &&
          submit && (
            <p className={styles.error}>
              *Todos los campos son obligatorios. Por favor complete los campos
              vacios.
            </p>
          )}
        <div className={styles.container_textarea}>
          <div>
            <h4 className={styles.subtitle}>Normas de entrega</h4>
            <label className={styles.label}>Descripción</label>
            <textarea
              className={styles.textarea_product_policy}
              onChange={handleInputChange}
              name="deliveryStandards"
              placeholder="Escribir"
              value={productPolicy.deliveryStandards}
            />
          </div>
          <div>
            <h4 className={styles.subtitle}>Salud y seguridad</h4>

            <label className={styles.label}>Descripción</label>
            <textarea
              className={styles.textarea_product_policy}
              onChange={handleInputChange}
              name="healthAndSecurity"
              placeholder="Escribir"
              value={productPolicy.healthAndSecurity}
            />
          </div>
          <div>
            <h4 className={styles.subtitle}>Política de cancelación</h4>
            <label className={styles.label}>Descripción</label>
            <textarea
              className={styles.textarea_product_policy}
              onChange={handleInputChange}
              name="cancellationPolicy"
              placeholder="Escribir"
              value={productPolicy.cancellationPolicy}
            />
          </div>
        </div>

        <div className={styles.btn_container}>
          <div className={styles.btn}>
            <Button
              type="submit"
              title="Modificar"
              btnStyle="btn--full"
              btnSize="btn--responsive"
              onClick={onSubmit}
            />
          </div>
        </div>
      </form>
    </div>
  );
}

export default ModifyProductPolicySection;
