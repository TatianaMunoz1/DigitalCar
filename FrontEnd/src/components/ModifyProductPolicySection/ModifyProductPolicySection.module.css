.container {
  background-color: #000000;
  height: 100%;
  padding: 15px;
}
.container_header {
  background: #000;
  position: absolute;
  top: 0px;
  left: 0px;
  height: 90px;
  width: 100%;
}
.error {
  color: rgb(255, 0, 0);
  padding-bottom: 15px;
  font-size: 12px;
  font-weight: 500;
}
.label {
  display: block;
  width: 100%;
  font-weight: 500;
  font-size: 14px;
  line-height: 20px;
}
.subtitle {
  color: #000000;
  font-size: 18px;
  line-height: 25px;
}
.container {
  background: #f0f0f0;
}
.container_textarea {
  display: grid;
  grid-template-rows: 1fr 1fr 1fr;
  gap: 20px;
}
.textarea_product_policy,
.input_textarea {
  width: 100%;
  border-radius: 8px;
  font-size: 14px;
  padding: 5px;
  border: none;
  resize: none;
  box-shadow: 0px 1px 5px rgba(0, 0, 0, 0.15);
}
.textarea_product_policy {
  height: 110px;
}
.input_textarea {
  border-radius: 8px;
  font-size: 14px;
  padding: 5px;
  border: none;
  resize: none;
  box-shadow: 0px 1px 5px rgba(0, 0, 0, 0.15);
}
.btn_container {
  width: 100%;
  display: flex;
  justify-content: center;
  padding-top: 50px;
}
.btn {
  max-width: 350px;
  width: 100%;
  align-self: center;
}

/*-----TABLET-----*/
@media (min-width: 760px) and (max-width: 1280px) {
  .container {
    padding: 30px;
    border-radius: 0px 8px 8px 8px;
  }
  .textarea_product_policy {
    height: 200px;
    padding: 20px;
  }
  .error {
    font-size: 14px;
  }
  .subtitle {
    color: #000000;
    font-size: 20px;
  }
  .btn_container {
    width: 100%;
    display: flex;
    justify-content: center;
    padding-top: 50px;
  }
  .btn {
    width: 350px;
    align-self: center;
  }
}

/*-----DESKTOP-----*/
@media (min-width: 1281px) {
  .title {
    color: #ffffff;
    font-size: 30px;
    line-height: 50px;
  }
  .subtitle {
    color: #000000;
    font-size: 20px;
  }
  .container {
    border-radius: 0px 8px 8px 8px;
    padding: 50px 30px;
  }
  .error {
    font-size: 18px;
  }
  .container_textarea {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr;
    grid-template-rows: 1fr;
    gap: 30px;
    padding: 0px;
  }
  .textarea_product_policy {
    height: 400px;
    padding: 10px;
    font-size: 15px;
    line-height: 24px;
  }
  .btn_container {
    width: 100%;
    display: flex;
    justify-content: center;
    padding-top: 50px;
  }
  .btn {
    width: 350px;
    align-self: center;
  }
}
