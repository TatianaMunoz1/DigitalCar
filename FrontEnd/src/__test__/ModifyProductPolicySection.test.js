import React from "react";
import { mount, shallow } from "enzyme";
import "@testing-library/jest-dom/extend-expect";
import { render, screen } from "@testing-library/react";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import ModifyProductPolicySection from "../components/ModifyProductPolicySection/ModifyProductPolicySection";


describe("Modify Product Policy", () => {

    test("render Modify Product Policy", () => {
        const wrapper = shallow(<ModifyProductPolicySection/>)
        expect(wrapper).toMatchSnapshot();
    });
});