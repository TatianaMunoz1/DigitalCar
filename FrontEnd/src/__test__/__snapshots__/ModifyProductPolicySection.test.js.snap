// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modify Product Policy render Modify Product Policy 1`] = `ShallowWrapper {}`;
